src/app/InteractionModelEngine.cpp

```diff
@@ -759,6 +759,15 @@ bool InteractionModelEngine::CanEstablishReadTransaction(const ReadHandler * apR
     size_t activeReadHandlersOnCurrentFabric = 0;
 
     // It is safe to use & here since this function will be called on current stack.
+    if (apReadHandler->GetAttributePathCount() > kReservedPathsPerReadRequest ||
+        apReadHandler->GetEventPathCount() > kReservedPathsPerReadRequest ||
+        apReadHandler->GetDataVersionFilterCount() > kReservedPathsPerReadRequest)
+    {
+        // Doesn't matter how many other reads are in progress; we are not
```

Comment:
So, this doesn't quite align with what the spec is saying:

_"If there are not enough resources to support the number of paths in the action and the number of paths in the action exceeds the number of paths that is guaranteed to be supported for a Read Interaction"_

That first bit is not being up-held here with this logic.

It's important we not alter the interpretation of the minima into a maxima at all times - that pivot'ed interpretation should only apply when we have insufficient resources to serve such a request.

Reply:
> That first bit is not being up-held here with this logic.

I am not changing the behavior of how many paths we allow.  Right now our behavior is to only allow the spec-defined minimum, so if you send any more paths than that the read is rejected.  That's true both with my change and without it.  The only thing my change does is change which status code is reported.

Now I agree that we should allow more paths if more paths are available, especially if we don't have the maximum number of fabrics commissioned.  But that's out of scope for this PR, which is just making sure that clients can tell when they need to retry vs when retrying the same thing would be pointless...

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/18610 to track this.